src/reducer/auth-reducer.js

```diff
@@ -0,0 +1,53 @@
+export const userInitialState = {
+  id: null,
+  firstName: "",
+  lastName: "",
+  email: "",
+  isAuthTokenPresent: false,
+  authToken: "",
+  msg: "",
+  likes: [],
+  history: [],
+  playlists: [],
+};
+
+export const authReducer = (state, action) => {
+  switch (action.type) {
+    case "LOGGED_IN": {
+      const { firstName, lastName, _id, email } = action.payload;
+      const token = action.token;
+      return {
+        ...state,
+        id: _id,
+        firstName: firstName,
+        lastName: lastName,
+        email: email,
+        isAuthTokenPresent: true,
+        authToken: token,
+      };
+    }
+    case "SIGNUP": {
+      const { firstName, lastName, _id, email } = action.payload;
+      const token = action.token;
+      return {
+        ...state,
+        id: _id,
+        firstName: firstName,
+        lastName: lastName,
+        email: email,
+        isAuthTokenPresent: true,
+        authToken: token,
+      };
```

Comment:
suggestion/can be done later: If you look closely, the two cases are almost the same, so you can have just one case for login/signup. Do play around with this.

Reply:
Okay